src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -4602,7 +4602,7 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                     AST::ConstantStr_t *n = AST::down_cast<AST::ConstantStr_t>(at->m_value);
                     std::string res = n->m_value;
                     int ind = 0;
-                    while (ind < res.size() && res[ind] == ' ') {
+                    while (ind < (int)res.size() && res[ind] == ' ') {
```

Comment:
Instead of casting `res.size()` to `int` can you try the following,

`size_t ind = 0;`

This review applies elsewhere in your PR as well.

Reply:
it makes variables unsigned, so it raise warinng like this 
```bash
warning: comparison of unsigned expression in ‘>= 0’ is always true [-Wtype-limits]
  700 |         while (ind >= 0 && res[ind--] == ' ');
      |                ~~~~^~~~
```
to fix this i will casting `ind` when i compare it with zero `ind >=0 `, so it's the same as casting the `size` it self i think.
am i right?

Reply:
Makes sense. @certik Should we merge this then?